torch/testing/_asserts.py

```diff
@@ -154,49 +154,37 @@ def _equalize_attributes(a: torch.Tensor, b: torch.Tensor) -> Tuple[torch.Tensor
     return a, b
 
 
-_Trace = namedtuple("Trace", ("total", "abs", "rel", "idx", "diff", "a", "b"))
-
-
-def _trace_mismatches(a: torch.Tensor, b: torch.Tensor, mismatches: torch.Tensor) -> _Trace:
-    """Traces mismatches and returns the found information.
-
-    The returned named tuple has the following fields:
-    - total (int): Total number of values in :attr:`a` and :attr:`b`.
-    - abs (int): Absolute number of mismatches.
-    - rel (float): Relative number of mismatches.
-    - idx (Union[int, Tuple[int, ...]]): Index of greatest absolute difference.
-    - diff (Union[int, float]): Greatest absolute difference.
-    - a (Union[int, float]): Value of :attr:`a` at the greatest absolute difference.
-    - b (Union[int, float]): Value of :attr:`a` at the greatest absolute difference.
-
-    For ``diff``, ``a``, and ``b`` the returned type depends on the :attr:`~torch.Tensor.dtype` of :attr:`a` and
-    :attr:`b`.
+def _trace_mismatches(a: torch.Tensor, b: torch.Tensor, mismatches: torch.Tensor) -> str:
+    """Traces mismatches.
 
     Args:
         a (torch.Tensor): First tensor.
         b (torch.Tensor): Second tensor.
         mismatches (torch.Tensor): Boolean mask of the same shape as :attr:`a` and :attr:`b` that indicates the
             location of mismatches.
+
+    Returns:
+        (str): Multiline list comprising the absolute and relative number of mismatches, and the maximum absolute and
+            relative difference with the corresponding index.
     """
-    total = mismatches.numel()
-    abs = torch.sum(mismatches).item()
-    rel = abs / total
+    total_elements = mismatches.numel()
+    abs_mismatches = torch.sum(mismatches).item()
+    rel_mismatches = abs_mismatches / total_elements
 
     dtype = torch.float64 if a.dtype.is_floating_point else torch.int64
     a_flat = a.flatten().to(dtype)
     b_flat = b.flatten().to(dtype)
 
-    abs_diff_flat = torch.abs(a_flat - b_flat)
-    idx_flat = torch.argmax(abs_diff_flat)
-
-    return _Trace(
-        total=total,
-        abs=abs,
-        rel=rel,
-        idx=_unravel_index(idx_flat, a.shape),
-        diff=abs_diff_flat[idx_flat].item(),
-        a=a_flat[idx_flat].item(),
-        b=b_flat[idx_flat].item(),
+    abs_diff = torch.abs(a_flat - b_flat)
+    max_abs_diff, max_abs_diff_flat_idx = torch.max(abs_diff, 0)
+
+    rel_diff = abs_diff / torch.abs(b_flat)
```

Comment:
This is an interesting way of computing the greatest relative difference, because it ignores the magnitude of elements of a. 

What if a is large but b is small? Won't that report a huge relative error but if a is small and b is large the relative error would be small?

Can we consider being symmetric? I understand that NumPy's actual comparison operations are asymmetric. I wonder if that's changing in the array API?

Reply:
Note, that [`torch.isclose`](https://pytorch.org/docs/stable/generated/torch.isclose.html?highlight=isclose#torch.isclose) and thus in turn `assert_tensors_close` is also asymmetric. This way `rel_diff` is the minimal value for `rtol` that would make the check pass.

> Can we consider being symmetric?

Sure, how would you go about it? Compute the relative difference both ways and pick the smaller one?

> I understand that NumPy's actual comparison operations are asymmetric.

Not sure what you mean by "actual comparison". If you have a look at [`numpy.testing.assert_allclose`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy-testing-assert-allclose) they are using the same logic to find mismatches. They named the parameters `actual` and `desired` to make this more clear. 

I'll check if they do anything different for their error message.

Reply:
You're right. I guess if we keep it this way then the atol and rtol are just these values, because the check is consistent with close.

What about "actual" and "expected" for names? It'd be nice to clarify to users which tensor should go where.

Reply:
Can do that. Do you want this naming scheme only for `assert_close` or everywhere for consistency? `assert_equal` is symmetric.

Reply:
What do you think?

Reply:
I would do it everywhere for consistency. Since the first two inputs should be positional only (although they can't be right now), this should not make it more verbose for the users. Plus, in case we want something asymmetric for `assert_equal` we are on the safe side.